### PR TITLE
Rule: add noUselessFragment rule and test

### PR DESCRIPTION
**Pull request by VictorHom**
[rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-useless-fragment.md)

#341

### CODE REVIEW

File: packages/@romejs/compiler/lint/rules/react/noUselessFragment.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	enter(path: Path): TransformExitResult {
+		const {node, context} = path;
+
+		if (node.type !== "JSXFragment" && !isJSXElement(node)) {
+			return node;
+		}
+
+		if (
+			path.parentPath.node.type !== "JSReturnStatement" &&
+			path.parentPath.node.type !== "JSVariableDeclarator" &&
```
Comment (sebmck) [inline packages/@romejs/compiler/lint/rules/react/noUselessFragment.ts:57]: You should just be able to use `path.parent` here

```suggestion
			path.parent.type !== "JSReturnStatement" &&
			path.parent.node.type !== "JSVariableDeclarator" &&
```

Comment (VictorHom) [reply packages/@romejs/compiler/lint/rules/react/noUselessFragment.ts:57]: I can make the change. When should one be used over the other? (so I know how to think about it in the future)

Comment (VictorHom) [reply packages/@romejs/compiler/lint/rules/react/noUselessFragment.ts:57]: Lol path.parent just points to path.ParentPath.node. NVM

**Comment by VictorHom** [comment]
@sebmck I have a question about. So I am seeing failures and I believe the failures are because the rule is being applied to `jsxFragments` rule.

On the one hand, it could be a matter of just:
1. removing those tests (which could possibly degrade the test and the unit cases it's testing)
2. accepting those tests changes (which could make it somewhat unclear to a contributor who looks at the result for the first time and can't see that another rule is being applied)
3. if there's a way to not apply the rule (which lets us keep both coverage and clarity)

What are your thoughts?

**Comment by sebmck** [comment]
@VictorHom I think we should just update the `jsxFragments` tests to include multiple inner children.

> 3. if there's a way to not apply the rule (which lets us keep both coverage and clarity)

So far we've deliberated avoiding being specific with the rules for test runs. It makes the execution ordering consistent with what will actually be ran for the user and acts closer to an integration test.

**Pull request merged**